### PR TITLE
Creating git commands with git-* pattern

### PR DESCRIPTION
**Pull request by kidd**
Also adding them to zsh's completion system.

### CODE REVIEW

**Comment by vladfr** [comment]
+1

Cool, I always wondered about this! Bonus points for zsh completion.

**Comment by kidd** [comment]
There are much more complex ways to add the completion , like add also documentation line for each command, but then you have to add it for every command.... and you'll end un forgetting it :)

**Comment by pirj** [comment]
Perfect!

**Comment by maurogeorge** [comment]
:star: good to know that.

I don't know it is valid to mention, but I can obtain the same behavior with a alias.

Via the command line:

```
$ git config --global alias.extend commit --amend --no-edit
```

Or directly on your .gitconfig

```
[alias]
  extend = commit --amend --no-edit
```

**Comment by dideler** [comment]
This is pretty cool!

@maurogeorge git aliases are good for one-liners. The method described in this TIL seems good for more complicated things that would be overkill for an alias. (I would use an alias for the example in the TIL.)

**Comment by maurogeorge** [comment]
@dideler :+1:

**Comment by kidd** [comment]
Thanks for the props!

**Comment by JoelQ** [comment]
@thoughtbot/shell thoughts here?

**Comment by Keithbsmiley** [comment]
Agree with @dideler. I don't think we should recommend making an executable for this specific example. Probably worth using something more complicated (like something from [here](https://github.com/garybernhardt/dotfiles/tree/master/bin)) as an example for this case. I do think it's worth documenting both cases.

**Comment by JoelQ** [comment]
Keep in mind that TILs are short learnings, not complete tutorials. I probably wouldn't cover both `git-*` and `alias` in the same TIL.

@dideler @Keithbsmiley agreed that something a little more complex might be a better example for the `git-*` executable custom command.

**Comment by kidd** [comment]
I used this simple example because the important part of the entry is not the command itself. more complex ones would mean explaining them which would go against the 'one thing per entry'.

File: git/add-own-git-commands.md [hunk 1]
```diff
@@ -0,0 +1,24 @@
+# Adding your own git commands
+
+Git uses a simple schema for all its commands. Any executable in your
+$PATH named git-foo, will be reachable via `git foo` from the
```
Comment (mike-burns) [inline git/add-own-git-commands.md]: Put `$PATH` in backticks, please.

**Pull request closed without merging**